### DISCUSSION TITLE
Repatriate rust_decimal implementation

### DISCUSSION THREAD

**Discussion by Sytten** [diesel-rs/diesel#3697] [Ideas]
While working on the sea-query/diesel bridge, I found out that `rust_decimal` implementations `ToSql` and `FromSql` are in the `rust_decimal` crate. I find that bit unintuitive so I was wondering if we were interested in having our own implementations and feature flag.

Another motivation is that the way they do feature flagging (`rust_decimal/db-diesel2-mysql`, `rust_decimal/db-diesel2-postgres`) makes it annoying to provide a single feature in depending crate. I guess you can use the `?`, but there is a notorious bug with (https://github.com/rust-lang/cargo/issues/10801) and I tend to avoid it.

**Comment by weiznich** [comment]
I generally try to avoid adding more integrations to diesel itself, as at least I personally do not have capacity to maintain much more code. If someone else maintains the code in the corresponding crate, that's from my personal point of view a "better" solution for the maintenance problem.

**Reply by Sytten** [reply]
Thats fair, it really is annoying that we can't create a bunch of community repositories to maintain those integrations without them being in the core or third party. Hopefully we see relaxation to the orphan rule at some point.
